core/state_processor.go

```diff
@@ -105,9 +105,15 @@ func ApplyTransaction(config *params.ChainConfig, bc *BlockChain, author *common
 
 	// Update the state with pending changes
 	usedGas.Add(usedGas, gas)
+	var root []byte
+	if config.IsMetropolis(header.Number) {
+		statedb.Finalise()
+	} else {
+		root = statedb.IntermediateRoot(config.IsEIP158(header.Number)).Bytes()
+	}
 	// Create a new receipt for the transaction, storing the intermediate root and gas used by the tx
 	// based on the eip phase, we're passing wether the root touch-delete accounts.
-	receipt := types.NewReceipt(statedb.IntermediateRoot(config.IsEIP158(header.Number)).Bytes(), usedGas)
+	receipt := types.NewReceipt(root, usedGas)
```

Comment:
Since the option 1 of EIP98 was chosen, so it's not enough to make `root` the empty sequence.  The receipt has one less elements in Metropolis.  See [a log of a coredev meeting](https://github.com/ethereum/pm/blob/master/All%20Core%20Devs%20Meetings/Meeting%2014.md#3-eip-98-removing-medstate-from-receipts---go--cpp-implemented-option-2-parity-seems-to-have-went-with-option-1-andrei).